Modules/Core/Common/src/itkMetaDataDictionary.cxx

```diff
@@ -53,29 +52,36 @@ void
 MetaDataDictionary
 ::Print(std::ostream & os) const
 {
+  os << "Dictionary use_count: " << m_Dictionary.use_count() << std::endl;
   for ( MetaDataDictionaryMapType::const_iterator it = m_Dictionary->begin();
         it != m_Dictionary->end();
         ++it )
     {
     os << ( *it ).first <<  "  ";
     ( *it ).second->Print(os);
     }
+
 }
 
 MetaDataObjectBase::Pointer &
 MetaDataDictionary
 ::operator[](const std::string & key)
 {
+  MakeUnique();
   return ( *m_Dictionary )[key];
 }
 
 const MetaDataObjectBase *
 MetaDataDictionary
 ::operator[](const std::string & key) const
 {
-  MetaDataObjectBase::Pointer entry = ( *m_Dictionary )[key];
-  const MetaDataObjectBase *  constentry = entry.GetPointer();
+  auto iter = m_Dictionary->find(key);
```

Comment:
@N-Dekker I pondered deprecating this method, but instead I have just made it return a nullptr, and not modify the std::map when not does not exist. This const method does not exist in the std container, and the behavior is different than the non-constant method.